### ISSUE TITLE
file reference support

### ISSUE THREAD

**Issue by teidesu** [mtcute/mtcute#40]
https://github.com/tdlib/td/blob/56163c2460a65afc4db2c57ece576b8c38ea194b/td/telegram/FileReferenceManager.cpp

implement file reference storage like described here: https://core.telegram.org/api/file_reference

**Comment by marioparaschiv** [comment]
Hey! I'm loving this library so far. I switched over from gramjs as the api is a nightmare to learn compared to mtcute.

Are there any updates on this issue?

I'm building a very attachment-heavy application and encounter expired file references often.

**Comment by teidesu** [comment]
not really, correctly handling them is a pita, and for most cases the fix is to just refetch the message

**Comment by teidesu** [comment]
i didn't really understand what you wrote

**Comment by h3hheewe** [comment]
his idea is good tho, as for owner of this repo there's certain cases where you simply can't refetch data
